### ISSUE TITLE
Arc commands don't get whitespace trimmed

### ISSUE THREAD

**Issue by launchpad-issue-importer** [scour-project/scour#89]
Most path commands are intelligently stripped so negative numbers, for instance, are not redundantly preceded by whitespace, but not arcs:

Example svg (the Twitter logo) that scour doesn't touch:

```
 <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 335 276" fill="#3ba9ee">
  <path d="m302 70a195 195 0 0 1 -299 175 142 142 0 0 0 97 -30 70 70 0 0 1 -58 -47 70 70 0 0 0 31 -2 70 70 0 0 1 -57 -66 70 70 0 0 0 28 5 70 70 0 0 1 -18 -90 195 195 0 0 0 141 72 67 67 0 0 1 116 -62 1
 </svg>
```

Shorter version:

```
 <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 335 276" fill="#3ba9ee">
  <path d="m302 70a195 195 0 0 1-299 175 142 142 0 0 0 97-30 70 70 0 0 1-58-47 70 70 0 0 0 31-2 70 70 0 0 1-57-66 70 70 0 0 0 28 5 70 70 0 0 1-18-90 195 195 0 0 0 141 72 67 67 0 0 1 116-62 117 117 0 0
 </svg>
```

---

Imported from Launchpad using lp2gh.
- date created: 2013-01-13T01:05:31Z
- owner: ecmanaut
- the launchpad url was https://bugs.launchpad.net/bugs/1099008

**Comment by launchpad-issue-importer** [comment]
(by ecmanaut)

**Comment by Ede123** [comment]
This is "by design", see cf5fb8a37d345dd1e265d3bb7b4a05909415f446 ([direct link](https://bugs.launchpad.net/scour/+bug/412754) to the relevant bug).

According to Jeff *"The final coordinate pair in the elliptical arc command must have whitespace preceding it."*. I'm unsure about that, though...

According to SVG 1.1 spec the [elliptical arc command](https://www.w3.org/TR/SVG/paths.html#PathDataEllipticalArcCommands) has the following parameters:
`(rx ry x-axis-rotation large-arc-flag sweep-flag x y)+`

The [grammar](https://www.w3.org/TR/SVG/paths.html#PathDataBNF) for it reads
```
elliptical-arc-argument-sequence:
    elliptical-arc-argument
    | elliptical-arc-argument comma-wsp? elliptical-arc-argument-sequence
elliptical-arc-argument:
    nonnegative-number comma-wsp? nonnegative-number comma-wsp? 
        number comma-wsp flag comma-wsp? flag comma-wsp? coordinate-pair
```

So if I'm not blind to the problem there is only one required `comma-wsp` which is in front of the "large-arc-flag". However (since this flag can only have the values 0 or 1) Scour would always put a space there anyway.

@codedread On the off chance that you have a *really* good memory: Do you remember why you forced Scour to output spaces here?

**Comment by Ede123** [comment]
Ahhh, found the source of confusion: They changed the spec!

See [comment #2](https://bugs.launchpad.net/inkscape/+bug/412815) in an old Inkscape bug for Jeffs reason to force Scour to output spaces here. There the grammar reads
```
elliptical-arc-argument:
    nonnegative-number comma-wsp? nonnegative-number comma-wsp?
        number comma-wsp flag comma-wsp flag comma-wsp coordinate-pair
```
i.e. *all* of the three `comma-wsp` in the last line are mandatory.

Since this grammar was never part of the final SVG 1.1 specification and most renderers did never suffer from this issue I tend to drop the behavior altogether. Inkscape still renders this wrongly in 0.91 (the current stable version), though, but will finally have fixed it in the soon to be released Inkscape 0.92.

Anybody feels differently let me know, then we'll keep this workaround as a ``--renderer-workaround`?
